iOS Client/DevotionsController/DevotionsCoordinator.swift

```diff
@@ -36,45 +37,56 @@ class DevotionsCoordinator: NSObject, Coordinator, StoreSubscriber, UNUserNotifi
                         MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: newDevotions)))
-                        self.promptForNotifications()
+                        self.promptForNotifications(withDevotions: newDevotions)
                     }
                 }
             }
         } else {
             MBStore.sharedStore.dispatch(LoadedDevotions(devotions: Loaded.loaded(data: devotions)))
-            self.promptForNotifications()
+            self.promptForNotifications(withDevotions: devotions)
         }
     }
     
-    func promptForNotifications() {
-        let scheduled = UserDefaults().bool(forKey: "scheduledNotifications")
-        if !scheduled{
-            let center = UNUserNotificationCenter.current()
-            center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
-                if granted {
-                    let devotions = self.devotionsStore.getDevotions()
-                    self.scheduleNotifications(devotions: devotions)
-                    UserDefaults().set(true, forKey: "scheduledNotifications")
-                } else {
-                    print(error)
-                }
+    func promptForNotifications(withDevotions devotions: [LoadedDevotion]) {
+        let center = UNUserNotificationCenter.current()
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+            if granted {
+                self.scheduleNotifications(withCenter: center, forDevotions: devotions)
+            } else {
+                print(error)
             }
         }
     }
     
-    func scheduleNotifications(devotions: [LoadedDevotion]) {
+    func scheduleNotifications(withCenter center: UNUserNotificationCenter, forDevotions devotions: [LoadedDevotion]) {
         print("scheduling notifications")
-        let center = UNUserNotificationCenter.current()
-        center.removeAllPendingNotificationRequests()
-        
-        devotions.forEach { devotion in
-            guard let dateComponents = devotion.dateComponentsForNotification else {
+
+        center.getPendingNotificationRequests { (requests) in
+            let startDate = Date().toMMddString()
+            let sortedDevotions = devotions.sorted {$0.date < $1.date}
+            guard let startIndex = (sortedDevotions.index { $0.dateAsMMdd == startDate }) else {
                 return
             }
-            let trigger = UNCalendarNotificationTrigger(dateMatching: dateComponents, repeats: true)
-            let content = DevotionNotificationContent(devotion: devotion)
-            let request = UNNotificationRequest(identifier: UUID().uuidString, content: content, trigger: trigger)
-            center.add(request)
+            
+            var i = startIndex
+            var outstandingNotifications: Int = 0
+            
+            while outstandingNotifications < MBConstants.DEVOTION_NOTIFICATION_WINDOW_SIZE {
```

Comment:
Was trying to think of a way to make this more swifty, but I think it gets the job done

Reply:
I'M_ALL_ABOUT_THAT_C++ lol